lib/style-property-matcher.js

```diff
@@ -121,15 +121,26 @@ var StylePropertyMatcher = (function (window, document) { // jshint ignore:line
      * Class that matches element style against the specified expression
      */
     var Matcher = function (propertyFilter, pseudoElement) {
-
         var propertyName;
         var regex;
 
         try {
-            var parts = propertyFilter.split(":", 2);
-            propertyName = parts[0].trim();
-            var regexText = SimpleRegex.createRegexText(parts[1].trim());
-            regex = new RegExp(regexText, "i");
+            var index = propertyFilter.indexOf(":");
+            propertyName = propertyFilter.substring(0, index).trim();
+            var pattern = propertyFilter.substring(index + 1).trim();
+
+            // Unescaping pattern
+            // For non-regex patterns, (,),[,] should be unescaped, because we require escaping them in filter rules.
+            // For regex patterns, ",\ should be escaped, because we manually escape those in style-property-matcher.js.
+            if(/^\/.*\/$/.test(pattern)) {
+                pattern = pattern.slice(1,-1).replace(/\\(["\\])/g, '$1');
+                regex = new RegExp(pattern);
+            }
+            else {
+                pattern = pattern.replace(/\\([()[\]])/g, '$1');
```

Comment:
Here it is unescaping escaped `(, ),\`s only. Should we unescape other characters too?

For example, `:contains(a\bc)` will be the same as `:contains(abc)`, because Sizzle unescapes `\b`, but currently `:matches-css(background-image: example.com)` will be different from `:matches-css(background-image: e\xample.com)`.

Maybe I should apply similar unescaping procedure to `:matches-css` for consistency with `:contains`.